scipy/ndimage/_backend.py

```diff
@@ -0,0 +1,268 @@
+import scipy._lib.uarray as ua
+from scipy.ndimage import _api
+import numpy as np
+
+
+__all__ = ['register_backend', 'set_backend',
+           'set_global_backend', 'skip_backend']
+
+
+class ndimage_output:
+    """
+    Special case outputs to handle multiple types in __ua_convert__.
+    Eg: output='f', output='float32', output=np.float32
+        are all acceptable in addition to arrays.
+    """
+    pass
+
+
+class ndimage_index:
+    """
+    Special case index argument in measurement methods to
+    handle both int and array types in __ua_convert__.
+    """
+    pass
+
+
+class _ScipyImageBackend:
+    """The default backend for ndimage calculations
+
+    Notes
+    -----
+    We use the domain ``numpy.scipy`` rather than ``scipy`` because ``uarray``
+    treats the domain as a hierarchy. This means the user can install a single
+    backend for ``numpy`` and have it implement ``numpy.scipy.ndimage`` as
+    well.
+
+    .. versionadded:: 1.8.0
+    """
+    __ua_domain__ = "numpy.scipy.ndimage"
```

Comment:
I understand the intention and I know this is done the same way for `scipy.fft`, but it's weird to call the hypothetical top-level backend `numpy`.
uarray supports a list of domains that can be used when implementing a single backend that supports multiple domains https://github.com/Quansight-Labs/uarray/pull/243 and it's just easier to understand the domain hierarchy when there's no `"numpy."` at the beginning of the string.
@hameerabbasi, @peterbell10 what do you think, is it really necessary to have `numpy` here?